app/models/user.rb

```diff
@@ -80,6 +81,16 @@ def by_conference(conference)
 
   after_save :touch_events
 
+  after_create_commit :mailbluster_create_lead
+  after_destroy_commit :mailbluster_delete_lead
+  # Note that because a commit may cause multiple changes
+  # which are not fully tracked by ActiveRecord::Dirty,
+  # we must resort to using a ActiveRecord::Concern which accumulates
+  # changes from a commit (app/models/concerns/track_saved_changes.rb)
+  # See https://github.com/ccmcbeck/after-commit
+  after_update_commit :mailbluster_update_email, if: ->(obj){ obj.ts_saved_changes.key? 'email' }
```

Comment:
> Using both after_create_commit and after_update_commit in the same model will only allow the last callback defined to take effect, and will override all others.

https://guides.rubyonrails.org/active_record_callbacks.html

Does this affect this code at all? It seems like only one of these applies, but the comment is a little unclear.

Reply:
I was not aware of that, I will look into it.

Reply:
I also think you might be able to just consolidate these two methods into `update_mailbluster_user` with a trigger for name or email.  Though I still can't tell if this affects the `after_create_commit`, too. So maybe there just needs to be a `create_or_update_mailbluster_user` method if necessary? Slightly annoying if so, but seems workable (just needs to check for `new?` or some sort of attribute like that)

Reply:
Ah I see what the issue is. That's slightly annoying.